### PR TITLE
Add rotating machines, curve and grounding related classes to the data model

### PR DESCRIPTION
**Pull request by Esaotuelr**
# Description

Add rotating machines, curve and grounding related classes to the data model.

If you have added new dependencies to the project please state why.

# Associated tasks

https://app.clickup.com/t/6929263/DEV-752

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [x] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.
- [x] I have commented my code in any hard-to-understand or hacky areas.
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).

### Documentation
- [x] I have updated the changelog.
- [x] I have updated any documentation required for these changes.

# Breaking Changes
- [x] I have considered if this is a breaking change and will communicate it with other team members if so.

This is not a breaking change

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [DEV-751 Update evolve-grpc](https://app.clickup.com/t/6929263/DEV-751)
Task linked: [DEV-752 Data model change spec](https://app.clickup.com/t/6929263/DEV-752)

File: proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  /**
+  * The IdentifiedObject fields for this Curve.
+  */
+  IdentifiedObject io = 1;
+
+  /**
+  * The point data values that define this curve.
+  */
+  CurveData curveData = 2;
```
Comment (kgreav) [inline proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto:32]: should be a `repeated CurveData` btw

File: proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import "zepben/protobuf/cim/iec61970/base/core/IdentifiedObject.proto";
+import "proto/zepben/protobuf/cim/iec61970/base/core/CurveData.proto";
+
+/**
+ * A multi-purpose curve or functional relationship between an independent variable (X-axis) and dependent (Y-axis) variables.
+ */
+message Curve {
+
+  /**
```
Comment (kgreav) [inline proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto:24]: looks like your indentation is configured incorrectly - needs to be 4 spaces

Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto]: done

File: proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto (hunk 1 above)
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/core/Curve.proto]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachineKind.proto [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+option csharp_namespace = "Zepben.Protobuf.CIM.IEC61970.Base.Wires";
+package zepben.protobuf.cim.iec61970.base.wires;
+
+/**
+ * Synchronous machine type.
+ */
+message SynchronousMachineKind {
+
+    /**
+     * NOTE: This enum is wrapped in a message due to a c98 naming conflict with SinglePhaseKind.Y
```
Comment (kgreav) [inline proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachineKind.proto:21]: the naming conflict will be with UNKNOWN

File: proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachine.proto [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+     * Modes that this synchronous machine can operate in.
+     */
+    SynchronousMachineKind type = 21;
+
+    /**
+     * Current mode of operation.
+     */
+    SynchronousMachineKind operatingMode = 22;
+
+
```
Comment (kgreav) [inline proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachine.proto:139]: missing:
`repeated ReactiveCapabilityCurve reactiveCapabilityCurveMRIDs = 23;`

File: spec/ewb/IEC61970/Base/Core/Curve.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+name: Curve
+description: A multi-purpose curve or functional relationship between an independent variable (X-axis) and dependent (Y-axis) variables.
+attributes:
+- name: curveData
+  type: CurveData
+  description: The point data values that define this curve.
+ancestors:
+- IdentifiedObject
+descendants:
+- ReactiveCapabilityCurve
```
Comment (kgreav) [inline spec/ewb/IEC61970/Base/Core/Curve.yaml:10]: add newlines to end of all these files plz

**Review by kgreav — CHANGES_REQUESTED** [review]
still gotta check the yamls... but almost there

File: proto/zepben/protobuf/cim/iec61970/base/wires/GroundingImpedance.proto [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+import "proto/zepben/protobuf/cim/iec61970/base/wires/EarthFaultCompensator.proto";
+
+/**
+ * A fixed impedance device used for grounding.
+ */
+message GroundingImpedance {
+
+  /**
+   * The ConductingEquipment fields for this EarthFaultCompensator.
```
[the remark this answers is not in the document]
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/GroundingImpedance.proto]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/PetersenCoil.proto [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import "zepben/protobuf/cim/iec61970/base/wires/EarthFaultCompensator.proto";
+
+/**
+ * A variable impedance device normally used to offset line charging during single line faults in an ungrounded section of network.
+ */
+message PetersenCoil {
+
+  /**
+   * The ConductingEquipment fields for this EarthFaultCompensator.
```
[the remark this answers is not in the document]
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/PetersenCoil.proto]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/GroundingImpedance.proto [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+import "zepben/protobuf/cim/iec61970/base/wires/EarthFaultCompensator.proto";
+
+/**
+ * A fixed impedance device used for grounding.
+ */
+message GroundingImpedance {
+
+    /**
+     * The ConductingEquipment fields for this EarthFaultCompensator.
```
[the remark this answers is not in the document]
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/GroundingImpedance.proto:23]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/PetersenCoil.proto [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import "zepben/protobuf/cim/iec61970/base/wires/EarthFaultCompensator.proto";
+
+/**
+ * A variable impedance device normally used to offset line charging during single line faults in an ungrounded section of network.
+ */
+message PetersenCoil {
+
+    /**
+     * The ConductingEquipment fields for this EarthFaultCompensator.
```
[the remark this answers is not in the document]
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/PetersenCoil.proto:23]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/ReactiveCapabilityCurve.proto [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Reactive power rating envelope versus the synchronous machine's active power, in both the generating and motoring modes. For each active power value there
+ * is a corresponding high and low reactive power limit value. Typically there will be a separate curve for each coolant condition, such as hydrogen pressure.
+ * The Y1 axis values represent reactive minimum and the Y2 axis values represent reactive maximum.
+ */
+message ReactiveCapabilityCurve {
+
+    /**
+     * The ConductingEquipment fields for this EarthFaultCompensator.
```
[the remark this answers is not in the document]
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/ReactiveCapabilityCurve.proto:25]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachineKind.proto (hunk 3 above)
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachineKind.proto:21]: done

File: proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachine.proto (hunk 4 above)
Comment (Esaotuelr) [reply proto/zepben/protobuf/cim/iec61970/base/wires/SynchronousMachine.proto:139]: done

File: spec/ewb/IEC61970/Base/Core/Curve.yaml (hunk 5 above)
Comment (Esaotuelr) [reply spec/ewb/IEC61970/Base/Core/Curve.yaml:10]: done

**Review by kgreav — APPROVED** [review]

**Pull request merged**